Summary table UI changes

BrandonSharratt opened the pull request:
*Issue:*https://github.com/bcgov/entity/issues/21656

*Description of changes:*
Added many components from BOR and one fully new component to handle the changes in the above noted ticket


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

BrandonSharratt commented:
/gcbrun

BrandonSharratt commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://bcregistry-btr-dev--pr-185-2hnjxhxx.web.app

kialj876 commented on btr-web/btr-common-components/pnpm-lock.yaml:
```diff
@@ -1,432 +1,697 @@
-      vitest-environment-nuxt:
-        specifier: ^1.0.0
-        version: 1.0.0(@testing-library/vue@8.0.2(@vue/compiler-sfc@3.4.21)(vue@3.4.21))(@vue/test-utils@2.4.5)(h3@1.11.1)(rollup@4.13.0)(vite@5.2.0-beta.1)(vitest@1.3.1)(vue-router@4.3.0(vue@3.4.21))(vue@3.4.21)
+dependencies:
```
can you revert all the changes in here and just add the new dependency thats needed please? It's very hard to tell what is being updated in here.

We do dependency upgrades outside of regular tickets because it can introduce a lot of minor issues that are hard to find in review

BrandonSharratt replied:
Done it's fully reverted because the only thing that happened was a small dependency bump which was unneeded

kialj876 reviewed (commented):
Hey nice work this is looking pretty good brandon, but I notice the icon images are not rendering in the pr build? And I don't think the director text should by there (i.e. Direct control, Indirect control, and Significant influence control of the majority of directors) since it is the same information as the icons right?

Also noticed the jointly / in concert lists include their own name

BrandonSharratt commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://bcregistry-btr-dev--pr-185-2hnjxhxx.web.app

BrandonSharratt commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://bcregistry-btr-dev--pr-185-2hnjxhxx.web.app

BrandonSharratt commented:
> Hey nice work this is looking pretty good brandon, but I notice the icon images are not rendering in the pr build? And I don't think the director text should by there (i.e. Direct control, Indirect control, and Significant influence control of the majority of directors) since it is the same information as the icons right?
> 
> Also noticed the jointly / in concert lists include their own name

Thanks for the notes, I have removed the director text about percentage as yea it doesn't make sense, the Jointly / In concert did list their own name in the design but I agree that was probably just as an example so I've removed them. I've also fixed the icons.

kialj876 commented on btr-web/btr-main-app/components/person-info/Control.vue:
```diff
@@ -0,0 +1,374 @@
+  const inconcert = sharesInConcert
+  if (jointly.value.length > 0) {
+    rv.push({
+      label: `Acting jointly (${jointly.value.length + 1})`,
```
Thanks for the updates! One last thing here and below, the '+1' should be removed so the counts are correct

hfekete commented on btr-web/btr-common-components/plugins/vue-country-flag.ts:
```diff
@@ -0,0 +1,5 @@
+import CountryFlag from 'vue-country-flag-next'
```
Did we really needed it as global component, could we not just load it when needed ?

Doesnt need changing, just wondering about reasoning.

hfekete replied:
Also, were you not able to use existing component BcrosCountryFlag

BrandonSharratt replied:
We already had vue-country-flag as an import I mostly looked to copy what BoR was doing

hfekete commented on btr-web/btr-main-app/components/bcros/AddressDisplay.vue:
```diff
@@ -0,0 +1,34 @@
+<template>
+  <div data-cy="address-display">
+    <div
```
Its preferred to have text wrapped in `<p></p>`. Its semantically more correct

hfekete commented on btr-web/btr-main-app/components/bcros/AddressDisplay.vue:
```diff
@@ -0,0 +1,34 @@
+<template>
+  <div data-cy="address-display">
+    <div
+      v-for="addressLine, i in addressData"
```
`v-for="addressLine, i in addressData"`
==> `v-for="(addressLine, i) in addressData"`
I would use brackets here, it shows better variable assignement

patrickpeinanw commented on btr-web/btr-main-app/components/person-info/Citizenship.vue:
```diff
@@ -0,0 +1,28 @@
+<template>
+  <div class="w-2/3 mr-2" :class="countries.length === 1 ? 'flex justify-center' : 'grid gap-x-1 grid-cols-2'">
+    <UTooltip
+      v-for="elem in countries"
+      :key="elem.alpha_2"
+      :text="getName(elem.alpha_2) || elem.alpha_2"
+      :popper="{ arrow: true, placement: 'top' }"
+    >
+      <CountryFlag
+        :country="elem.alpha_2"
+        size="normal"
+        style="margin-top: -5px"
+      />
+    </UTooltip>
```
Canadian Permanent Resident has a special alpha_2 "CA_PR" and it won't be recognized by `<CountryFlag>` and the `getName` function. The Canadian flag should be rendered for both "CA" and "CA_PR". The tooltip text for "CA" should be "Canada (Citizen)". For "CA_PR" the tooltip should show "Canada (Permanent Resident)".

BrandonSharratt replied:
Good catch, changed.

hfekete commented on btr-web/btr-main-app/components/person-info/Citizenship.vue:
```diff
@@ -0,0 +1,28 @@
+<template>
+  <div class="w-2/3 mr-2" :class="countries.length === 1 ? 'flex justify-center' : 'grid gap-x-1 grid-cols-2'">
+    <UTooltip
```
we have BcrosTooltip already, that is wrapping and setting some style, can you use it here ?

hfekete commented on btr-web/btr-main-app/components/person-info/Control.vue:
```diff
@@ -0,0 +1,374 @@
+const prop = defineProps<{ item: SiSchemaType }>()
+const interests = SiSchemaToBtrBodsConverters.getInterests(prop.item)
+
+const INTERESTS_SHARES_JOINT = 'controlType.shares.actingJointly'
```
We have enum `ControlOfSharesDetailsE` for this

hfekete commented on btr-web/btr-main-app/components/person-info/Control.vue:
```diff
@@ -0,0 +1,374 @@
+
+const INTERESTS_SHARES_JOINT = 'controlType.shares.actingJointly'
+const INTERESTS_SHARES_CONCERT = 'controlType.shares.inConcertControl'
+const INTERESTS_VOTES_JOINT = 'controlType.votes.actingJointly'
```
`ControlOfVotesDetailsE` for votes

hfekete commented on btr-web/btr-main-app/components/person-info/Control.vue:
```diff
@@ -0,0 +1,374 @@
+const INTERESTS_SHARES_CONCERT = 'controlType.shares.inConcertControl'
+const INTERESTS_VOTES_JOINT = 'controlType.votes.actingJointly'
+const INTERESTS_VOTES_CONCERT = 'controlType.votes.inConcertControl'
+const INTERESTS_DIRECTORS_JOINT = 'controlType.directors.actingJointly'
```
`ControlOfDirectorsDetailsE` for director stuff

patrickpeinanw reviewed (changes requested):
The table looks nice! Just one comment for the country flag. 

Also, I noticed that, if we make changes to the Acting Jointly or In-Concert table (i.e., selecting or unselecting a person from the dropdown menu), the corresponding 'jointly' or 'in concert' accordion in the summary table will not update. I am now sure if this is an expected behavior in the design.

hfekete commented on btr-web/btr-main-app/components/person-info/Details.vue:
```diff
@@ -0,0 +1,78 @@
+<template>
+  <div class="information">
+    <!-- NB: if the UIcons are not wrapped in dev they render at differring sizes -->
```
I would potentially put` class="w-x h-y" ` if you have specific size they can be

hfekete commented on btr-web/btr-main-app/components/person-info/Details.vue:
```diff
@@ -0,0 +1,78 @@
+</script>
+
+<style lang="scss" scoped>
+.information > div:not(:first-child) {
```
I think we are trying to use tailwind, rather then css/scss but keep it as is now

hfekete reviewed (commented):
Nice work, I would like to see few comments acted on, but few are just NB for future

BrandonSharratt commented:
/gcbrun

BrandonSharratt commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://bcregistry-btr-dev--pr-185-2hnjxhxx.web.app

patrickpeinanw reviewed (approved):
Looks good to me. Thanks for the updates.

kialj876 reviewed (approved):
Nice work 👍
